graph-gateway/src/main.rs

```diff
@@ -486,3 +452,64 @@ async fn handle_metrics() -> impl axum::response::IntoResponse {
 fn graphql_error_response<S: ToString>(message: S) -> json::JsonResponse {
     json::json_response([], json!({"errors": [{"message": message.to_string()}]}))
 }
+
+/// Creates a new [`AuthContext`] from the given configuration.
+///
+/// This functions awaits the completion of the initial API keys and subscriptions fetches.
+fn init_auth_service(
+    payment_required: bool,
+    api_keys_http_client: reqwest::Client,
```

Comment:
nit: the http client isn't isolated to api keys
```suggestion
    http_client: reqwest::Client,
```

Reply:
I am differentiating the two HTTP clients, API keys and Subscriptions, as both may have different settings (e.g., connection timeout, request timeout, etc.).